anesthetic/plot.py

```diff
@@ -323,11 +323,29 @@ def kde_plot_1d(ax, data, *args, **kwargs):
     xmax = kwargs.pop('xmax', None)
     weights = kwargs.pop('weights', None)
     ncompress = kwargs.pop('ncompress', 1000)
+    percentile = kwargs.pop('percentile', '5sigma')
+    if isinstance(percentile, str):
+        sigmas = {'1sigma': 0.682689492137086,
+                  '2sigma': 0.954499736103642,
+                  '3sigma': 0.997300203936740,
+                  '4sigma': 0.999936657516334,
+                  '5sigma': 0.999999426696856}
+        percentile = sigmas[percentile]
+    if isinstance(percentile, float) or isinstance((percentile, int)):
+        if percentile < 0.5:
+            percentile = 1 - percentile
+        percentile = (1-percentile, percentile)
+        print(percentile)
+
+    if weights is not None:
+        data = data[weights != 0]
+        weights = weights[weights != 0]
+
     x, w = sample_compression_1d(data, weights, ncompress)
     kde = gaussian_kde(x, weights=w)
     p = kde(x)
     p /= p.max()
-    i = ((x < quantile(x, 0.999, w)) & (x > quantile(x, 0.001, w))) | (p > 0.1)
+    i = ((x > quantile(x, percentile[0], w)) & (x < quantile(x, percentile[1], w))) | (p > 0.1)
```

Comment:
`percentile` can now be either a `str`, a `float`, or a `tuple of float`.

`str`: specify desired sigma level, gets converted to corresponding percentile
`float`: percentile (symmetric)
`tuple`: left and right percentile (can be asymmetric)

Reply:
Quantile is probably more accurate (as it's a number between 0 and 1).

Reply:
`quantile` is already taken by the function... Change the function name to `get_quantile`?

Reply:
> Quantile is probably more accurate (as it's a number between 0 and 1).

Actually, aren't both quantile and percentile referring to the x-values and not the p-values? Should I go for `pvalue` or `p_value` or `plot_p_value`  or something like that?  (`plot_p_interval`, `plot_p_range`, ...?)

Reply:
None of these are great. We could just be consistent with [scipy](https://numpy.org/doc/stable/reference/generated/numpy.quantile.html), and call it q, as we are plotting the inner 'q'-quantiles.